lib/katello/tasks/reset.rake

```diff
@@ -1,29 +1,57 @@
 namespace :katello do
   namespace :reset_backends do
-    service_stop = "sudo /sbin/service %s status > /dev/null && sudo /sbin/service %s stop"
-    service_start = "sudo /sbin/service %s start"
+    # Switch to foreman-maintain when Pulp3 services are added to it
+    service_stop = "sudo /usr/bin/systemctl status %s > /dev/null && sudo /usr/bin/systemctl stop %s"
+    service_start = "sudo /usr/bin/systemctl start %s"
 
-    task :pulp do
-      SERVICES = %w(httpd pulp_workers pulp_resource_manager pulp_celerybeat).freeze
-      system(service_stop.gsub("%s", "mongod"))
+    task :pulp_legacy do
+      SERVICES = %w(httpd pulp_workers pulp_resource_manager pulp_celerybeat squid qpidd pulp_streamer).freeze
+
+      puts "\e[33mStarting Pulp2 Reset\e[0m\n\n"
 
       SERVICES.each { |s| system(service_stop.gsub("%s", s)) }
-      system("sudo rm -rf /var/lib/mongodb/pulp_database*")
-      system(service_start.gsub("%s", "mongod"))
-      sleep(10)
-      fail "Cannot migrate pulp database" unless system("sudo -u apache /usr/bin/pulp-manage-db")
+      system(service_start.gsub("%s", 'rh-mongodb34-mongod'))
+      system("mongo pulp_database --eval 'db.dropDatabase();'")
+      fail "\e[31mCannot Migrate Pulp2 Database\e[0m\n\n" unless system('sudo -u apache /usr/bin/pulp-manage-db')
 
       SERVICES.each { |s| system(service_start.gsub("%s", s)) }
-      puts "Pulp database reset."
+      puts "\e[32mPulp2 Database Reset Complete\e[0m\n\n"
+    end
+
+    task :pulp do
+      # Pull active workers from Redis to see how many we need to start up
+      worker_count = `redis-cli --scan --pattern "rq:workers:reserved-resource-worker*" | wc -l`.chomp.to_i
+
+      services = %w(redis pulpcore-api pulpcore-resource-manager pulpcore-content)
+      while worker_count > 0
+        services.push("pulpcore-worker@#{worker_count}")
+        worker_count -= worker_count
+      end
```

Comment:
using this: 
```
systemctl start 'pulpcore-worker@*' --all
```

i think is much simpler and works even if the workers are down